website/content/en/docs/cloud-providers/AWS/launch-templates.md

```diff
@@ -0,0 +1,236 @@
+default OS, Amazon Linux 2 (AL2), accepts shell scripts (bash commands). 
+
+[AWS calls data passed to an instance at launch time "user
+data".](https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/user-data.html#user-data-shell-scripts)
+
+In the default configuration, Karpenter uses an EKS optimized version of AL2 and passes the hostname of the Kubernetes API server, and a certificate. The EKS Optimized AMI includes a `bootstrap.sh` script which connects the instance to the cluster, based on the passed data. 
+
+Alternatively, you may reference AWS's [`bootstrap.sh`
+file](https://github.com/awslabs/amazon-eks-ami/blob/master/files/bootstrap.sh)
+when building a custom base image. 
+
+```
+#!/bin/bash
+/etc/eks/bootstrap.sh <my-cluster-name> \
+--kubelet-extra-args <'--max-pods=40'> \
+--b64-cluster-ca <certificateAuthority> \
+--apiserver-endpoint <endpoint> 
+--dns-cluster-ip <serviceIpv4Cidr>
+--use-max-pods false
+```
+
+Note, you must populate this command with live values. Karpenter will
+not change the user data in the launch template. 
+
+Encode using yaml function `!Base64` yaml function or `cat userdata.sh | base64 > userdata-encoded.txt` shell command. 
+
+**Bootstrap Script Parameters**
+
+The sample bootstrap script requires information to join the cluster.
+
+These values may be found using:
+```
+aws eks describe-cluster --name MyKarpenterCluster 
+```
+
+**Kubelet Arguments**
+
+Specifying max-pods can break Karpenter's binpacking logic (it has no way to know what this setting is). If karpenter attempts to pack more than this number of pods, the instance may be oversized, and additional pods will reschedule.
+
+## Situational Fields 
+
+Configure these values in response to a particular use case, such as nodes interacting with another AWS service, or using EBS storage on the node. 
+
+### Instance Profile - IAM
+
+The launch template must include an "instance profile" -- a set of IAM roles. 
+
+The instance profile must include all the permissions of the default Karpenter
+node instance profile. For example, permission to run containers and manage
+networking. See the default role, `KarpenterNodeRole`, in the full example
+below for more information. 
+
+### Storage
+
+Karpenter expects nothing of node storage. Configure as needed for your base
+image.
+
+### Security Groups - Firewall
+
+EKS configures security groups (i.e., instance firewall rules) automatically. 
+
+However, you may manually specify a security group. The security group must
+permit communication with EKS control plane. Outbound access should be
+permitted for at least: HTTPS on port 443, DNS (UDP and TCP) on port 53, and
+your subnet's network access control list (network ACL). 
+
+The security group must be associated with the virtual private cloud (VPC) of
+the EKS cluster.
+
+## Fields with Undefined Behavior
+
+These resources referenced by these fields are controlled by EKS/Karpenter, and not the launch template. 

```

Comment:
`These resources referenced by these`. I think you can just drop the first `These`